python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -87,6 +87,21 @@ def test_help_command(self):
         pydoc.render_doc(df.foo)
         pydoc.render_doc(df.take(1))
 
+    def test_drop(self):
+        df = self.spark.createDataFrame([("A", 50, "Y"), ("B", 60, "Y")], ["name", "age", "active"])
+
+        self.assertEqual(df.drop("active").columns, ["name", "age"])
+
+        self.assertEqual(df.drop("active", "nonexistent_column").columns, ["name", "age"])
+
+        self.assertEqual(df.drop("name", "age", "active").columns, [])
```

Comment:
Does this fail without your patch?

Reply:
No, it shouldn't. The patch is related to `df.drop(col*)` and not `df.drop(str*)`. The added assert is for testing regression which this method lacked before this patch.